packages/dotnet-sdk/src/TeamsFx.Test/TeamsBotSsoPromptTest.cs

```diff
@@ -0,0 +1,122 @@
+    }
+    
+    [TestMethod]
+    public void TeamsBotSsoPromptWithEmptySettingShouldFail()
+    {
+        Assert.ThrowsException<ArgumentNullException>(() => new TeamsBotSsoPrompt(fakeDialogId, null));
+    }
+    #endregion
+
+    #region BeginDialog
+    [TestMethod]
+    public async Task TeamsBotSsoPromptBeginDialogWithNoDialogContextShouldFail()
+    {
+        await Assert.ThrowsExceptionAsync<ArgumentNullException>(async () =>
+        {
+            var prompt = new TeamsBotSsoPrompt(fakeDialogId, teamsBotSsoPromptSettingsMock);
+            await prompt.BeginDialogAsync(null);
+        });
+    }
+
+    [TestMethod]
+    public async Task TeamsBotSsoPromptBeginDialogNotInTeamsShouldFail()
+    {
+        var ex = await Assert.ThrowsExceptionAsync<ExceptionWithCode>(async () =>
+        {
+            var prompt = new TeamsBotSsoPrompt(fakeDialogId, teamsBotSsoPromptSettingsMock);
+            var convoState = new ConversationState(new MemoryStorage());
+            var dialogState = convoState.CreateProperty<DialogState>("dialogState");
+
+            var adapter = new TestAdapter()
+                .Use(new AutoSaveStateMiddleware(convoState));
+
+            // Create new DialogSet.
+            var dialogs = new DialogSet(dialogState);
+            dialogs.Add(prompt);
+
+            var tc = new TurnContext(adapter, new Activity() { Type = ActivityTypes.Message, Conversation = new ConversationAccount() { Id = "123" }, ChannelId = "not-teams" });
+
+            var dc = await dialogs.CreateContextAsync(tc);
+
+            await prompt.BeginDialogAsync(dc);
+        });
+        Assert.AreEqual(ExceptionCode.ChannelNotSupported, ex.Code);
+    }
+
+    //[TestMethod]
+    //public async Task TeamsBotSsoPromptBeginDialogShouldSuccess()
+    //{
+    //    var loggerMock = new Mock<ILogger<TeamsBotSsoPrompt>>();
+    //    var ex = await Assert.ThrowsExceptionAsync<ExceptionWithCode>(async () =>
+    //    {
+    //        var prompt = new TeamsBotSsoPrompt(fakeDialogId, teamsBotSsoPromptSettingsMock, loggerMock.Object);
+    //        var convoState = new ConversationState(new MemoryStorage());
+    //        var dialogState = convoState.CreateProperty<DialogState>("dialogState");
+
+    //        var adapter = new TestAdapter()
+    //            .Use(new AutoSaveStateMiddleware(convoState));
+
+    //        // Create new DialogSet.
+    //        var dialogs = new DialogSet(dialogState);
+    //        dialogs.Add(prompt);
+
+    //        var tc = new TurnContext(adapter, new Activity() { Type = ActivityTypes.Message, Conversation = new ConversationAccount() { Id = "123" }, ChannelId = Bot.Connector.Channels.Msteams });
+
+    //        var dc = await dialogs.CreateContextAsync(tc);
+
+    //        await prompt.BeginDialogAsync(dc);
+    //    });
+    //    Assert.AreEqual(ExceptionCode.ChannelNotSupported, ex.Code);
+    //}
+
+    #endregion
```

Comment:
We may need to add some more success cases

Reply:
Also, not sure whether it's possible to add cases for data annotations in BotAuthenticationOptions

Reply:
Updated a success flow test case.

Reply:
More test cases can be left to another PR.